src/runtime/d3d12compute.cpp

```diff
@@ -86,40 +86,52 @@ class StackPrinter : public Printer<type, length> {
 static void *const user_context = NULL;
 //
 #if HALIDE_D3D12_TRACE
-// it's better to use StackPrinter<> instead of Printer<> (debug(ctx)) when
-// tracing calls because Printer<> calls 'halide_malloc()' which can fail;
-// there's even a test for it (generator_aot_cleanup_on_error), simulating
-// a halide_malloc failure, and user code should not affect debug-tracing
-#define trace(x) StackPrinter<4096, BasicPrinter>(x)
-
-static volatile ScopedSpinLock::AtomicFlag indent_lock = 0;
-static const char indent_pattern[] = "   ";
-static char indent[2048] = {};
-static int indent_end = 0;
-#define TRACEINDENT ((const char *)indent)
-#define TRACEPRINT(msg) trace(user_context) << TRACEINDENT << msg;
+static volatile ScopedSpinLock::AtomicFlag trace_indent_lock = 0;
```

Comment:
nit: probably worth adding a comment here about why we are explicitly avoiding StackPrinter

Reply:
I just deleted StackPrinter entirely and made Printer use the stack when things are small instead.